### PR TITLE
[REVIEW] Add support for `cudf.Index` to create Indexes

### LINKED ISSUE #4675: [FEA] Allow constructing `Index` objects via `Index()`
We should allow constructing index objects using `Index()`. This will basically involve adding `Index.__new__()` which constructs an instance of the appropriate subclass.

```python
In [47]: pd.Index(['a', 'b', 'c'])
Out[47]: Index(['a', 'b', 'c'], dtype='object')

In [48]: cudf.Index(['a', 'b', 'c'])
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-48-17d2b6cfde17> in <module>
----> 1 cudf.Index(['a', 'b', 'c'])

~/workspace/github/shwina/cudf/python/cudf/cudf/_libxx/table.pyx in cudf._libxx.table.Table.__init__()

~/workspace/github/shwina/cudf/python/cudf/cudf/core/column_accessor.py in __init__(self, data, multiindex, level_names)
     36             self._level_names = level_names
     37
---> 38         self._data = OrderedColumnDict(data)
     39         self.multiindex = multiindex
     40         self._level_names = level_names

ValueError: need more than 1 value to unpack

```

### PR DESCRIPTION
**Pull request by rgsl888prabhu**
closes #4675

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cudf/cudf/core/index.py [hunk 1]
```diff
@@ -58,9 +58,50 @@ def _to_frame(this_index, index=True, name=None):
...
+    0  0  0
+    1  1  1)
     """
 
+    def __new__(cls, data=None, dtype=None, name=None, **kwargs):
+
+        if name is None:
+            return as_index(data, dtype=dtype, **kwargs)
+        else:
+            return as_index(data, dtype=dtype, name=name, **kwargs)
```
Comment (galipremsagar) [inline python/cudf/cudf/core/index.py:100]: I think we can straightaway pass `name` to `as_index` without any checks, no?

```python
>>> import cudf
>>> cudf.core.index.as_index([1, 2, 3])
Int64Index([1, 2, 3], dtype='int64')
>>> cudf.core.index.as_index([1, 2, 3]).name
>>> cudf.core.index.as_index([1, 2, 3], name='fs')
Int64Index([1, 2, 3], dtype='int64', name='fs')
>>> cudf.core.index.as_index([1, 2, 3], name=None)
Int64Index([1, 2, 3], dtype='int64')
```

Comment (rgsl888prabhu) [reply python/cudf/cudf/core/index.py:100]: Think I made that change well before when there was a conflict, not any more. Nice catch.

**Comment by kkraus14** [comment]
As a follow up to this PR, we should align the Index constructor APIs to match Pandas and throw on any unsupported args. Additionally, we may want to add an alias or some form of handling for things like `Int64Index`, `UInt64Index`, `Float64Index`, etc.

**Pull request merged**